### DISCUSSION TITLE
Manipulate environment variable in a recipe and have it carry over to another recipe

### DISCUSSION THREAD

**Discussion by oivviors** [casey/just#2185] [Q&A]
Say I want to manipulate som environment variable in a just file.  I know I can do that at the top level and have it affect all recipes like so for instance:

`export PATH := justfile_directory() + "/node_modules/.bin" + ":" + env_var('PATH')`

Now however I'll have changed PATH for in all my recipes and that might not be what I want.

Is there I way I can  encapsulate this env var manipulation in a recipe that I can then have other recipes depend on? 

The way I understand it ordinary sh-recipes execute independently of one another so env var manipulation in one will not carry over to the next further down the dependency chain.

**Comment by casey** [comment]
This is a super common and very reasonable question, but unfortunately there's no good way to do this. `just` tries to be order-independent, so what one recipe does doesn't affect other recipes, and each recipe is executed with a fresh shell, so no environment variables are shared. As one option, you could pass the path as a dependency to the recipes that needed it, and export it:

```just
foo $PATH:

bar: (foo (justfile_directory() + "/node_modules/.bin" + ":" + env_var('PATH')))
```

But that's pretty verbose and weird.

**Reply by oivvio** [reply]
Thanks.
